cpp/src/io/parquet/stats_filter_helpers.cpp

```diff
@@ -48,12 +49,66 @@ std::reference_wrapper<ast::expression const> stats_expression_converter::visit(
   return expr;
 }
 
-std::reference_wrapper<ast::expression const> stats_expression_converter::visit(
+std::reference_wrapper<ast::expression const> stats_columns_collector::visit(
   ast::column_name_reference const& expr)
 {
   CUDF_FAIL("Column name reference is not supported in statistics AST");
 }
 
+std::reference_wrapper<ast::expression const> stats_columns_collector::visit(
+  ast::operation const& expr)
+{
+  using cudf::ast::ast_operator;
+  auto const operands = expr.get_operands();
+  auto const op       = expr.get_operator();
+
+  if (auto* v = dynamic_cast<ast::column_reference const*>(&operands[0].get())) {
+    // First operand should be column reference, second should be literal.
+    CUDF_EXPECTS(cudf::ast::detail::ast_operator_arity(op) == 2,
+                 "Only binary operations are supported on column reference");
+    CUDF_EXPECTS(dynamic_cast<ast::literal const*>(&operands[1].get()) != nullptr,
+                 "Second operand of binary operation with column reference must be a literal");
+    v->accept(*this);
+    // If this is a supported operation, mark the column as needed
+    if (op == ast_operator::EQUAL or op == ast_operator::NOT_EQUAL or op == ast_operator::LESS or
+        op == ast_operator::LESS_EQUAL or op == ast_operator::GREATER or
+        op == ast_operator::GREATER_EQUAL) {
+      _columns_mask[v->get_column_index()] = true;
+    } else {
+      CUDF_FAIL("Unsupported operation in Statistics AST");
+    }
+  } else {
+    // Visit the operands and ignore any output as we only want to build the column mask
+    std::ignore = visit_operands(operands);
+  }
+
+  return expr;
+}
+
+thrust::host_vector<bool> stats_columns_collector::get_stats_columns_mask() &&
+{
+  return std::move(_columns_mask);
+}
+
+std::vector<std::reference_wrapper<ast::expression const>> stats_columns_collector::visit_operands(
```

Comment:
Function only moved as-is from stats_expression_converter to this new base class.